### ISSUE TITLE
Cannot use SSM variables in the serverless.yaml file

### ISSUE THREAD

**Issue by alanschapira** [serverless-nextjs/serverless-next.js#2251]
### Issue Summary
When you add an environment variable to the serverless.yaml file, I am able to add strings for my values, but not references to SSM.

### Actual behavior
It does build and deploy correctly, but it does not use the SSM variable. It will instead consider the SSM reference as a string.

### Expected behavior
It should use the SSM variable as it works in regular serverless like this: https://www.serverless.com/framework/docs/providers/aws/guide/variables#reference-variables-using-the-ssm-parameter-store

### Steps to reproduce
Putting this in the serverless.yaml file works:
```
build:
      env:
        MY_ENV_VAR: 'env_var_value'
```

but this does not work:
```
build:
      env:
        MY_ENV_VAR: '${ssm:/path/to/service/myParam}'
```

### Screenshots/Code/Configuration/Logs
N/A

### Versions

- OS/Environment: MacOS
- @sls-next/serverless-component version: 3.4.0
- Next.js version: 11.1.2

### Additional context

I am unsure if this is actually a bug, or a feature request because I am unsure if this is expected behaviour or not. I am unsure of the relationship between serverless-nextjs and serverless and if features from serverless are available on serverless-nextjs or if it is just a custom tool that creates some serverless config.

### Checklist

- [x] You have reviewed the [README](https://github.com/serverless-nextjs/serverless-next.js/blob/master/README.md) and [FAQs](https://github.com/serverless-nextjs/serverless-next.js#faq), which answers several common questions.
- [x] You have reviewed our [DEBUGGING](https://github.com/serverless-nextjs/serverless-next.js/wiki/Debugging-Issues) wiki and have tried your best to include complete information and reproduction steps (including your configuration) as is possible. As there is only one maintainer (who maintains this in his free time) and thus very limited resources, if you have time, please try to debug the issue a bit yourself if possible.
- [x] You have first tried using the most recent `latest` or `alpha` `@sls-next/serverless-component` [release version](https://github.com/serverless-nextjs/serverless-next.js/releases), which may have already fixed your issue or implemented the feature you are trying to use. Note that the old `serverless-next.js` component and the `serverless-next.js` plugin are deprecated and no longer maintained.

**Comment by anil1712** [comment]
@alanschapira I am also looking same, did you get any solution for it?

**Comment by alanschapira** [comment]
No, but it has only been a few days. My current solution is to have `env.ssm` file which has the keys of all my env variables and the values are pointers in SSM. e.g.

```
ENV_VAR_1="/path/to/env/var1"
ENV_VAR_2="/path/to/env/var2"
```

Then part of the deployment process, I run a script that will read a file and replace the values with the actual SSM value and stick it into a .env file.

Happy to share the script with you if interested. I have set it up to work with multiple environments so you can deploy to staging/prod environments easily. It also works locally for setting up your .env file for developers which is nice too.

**Comment by anil1712** [comment]
Thanks @alanschapira for responding. Yes please share the script with me.

**Comment by alanschapira** [comment]
Ive spent some time adding in details of how it works and how to set it up, but its up now [here](https://gist.github.com/alanschapira/cb8b2db46e1e15e893f3f2c0830b779b). Let me know if you have any questions about it! @anil1712

**Comment by anil1712** [comment]
@alanschapira Very very thanks to you for sharing the code.
